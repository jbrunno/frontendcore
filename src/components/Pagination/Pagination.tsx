import React from 'react';
import {
  Pagination as MuiPagination,
  PaginationProps as MuiPaginationProps,
} from '@mui/material';

export type PaginationProps = MuiPaginationProps;

export function Pagination(props: PaginationProps) {
  return <MuiPagination {...props} />;
}
export default Pagination;
