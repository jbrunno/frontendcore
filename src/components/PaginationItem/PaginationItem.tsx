import React from 'react';
import {
  PaginationItem as MuiPaginationItem,
  PaginationItemProps as MuiPaginationItemProps,
} from '@mui/material';

export type PaginationItemProps = MuiPaginationItemProps;

export function PaginationItem(props: PaginationItemProps) {
  return <MuiPaginationItem {...props} />;
}
export default PaginationItem;
