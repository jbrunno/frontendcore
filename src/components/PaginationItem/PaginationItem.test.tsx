import React from 'react';
import { render } from '@test-utils/component-testing';
import {
  ArrowBack as ArrowBackIcon,
  ArrowForward as ArrowForwardIcon,
} from '@mui/icons-material';
import { PaginationItem } from './PaginationItem';
import Pagination from '../Pagination/Pagination';

test('renders the PaginationItem', () => {
  const { container } = render(
    <Pagination
      count={10}
      renderItem={(item) => (
        <PaginationItem
          components={{ previous: ArrowBackIcon, next: ArrowForwardIcon }}
          {...item}
        />
      )}
    />,
  );
  expect(container).toMatchSnapshot();
});
