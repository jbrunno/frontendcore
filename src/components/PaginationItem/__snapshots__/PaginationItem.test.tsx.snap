// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PaginationItem 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  margin: 0;
  list-style: none;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  border-radius: 16px;
  text-align: center;
  box-sizing: border-box;
  min-width: 32px;
  height: 32px;
  padding: 0 6px;
  margin: 0 3px;
  color: #3B3B41DE;
  -webkit-transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2.Mui-focusVisible {
  background-color: #0000001F;
}

.emotion-2.Mui-disabled {
  opacity: 0.38;
}

.emotion-2:hover {
  background-color: #0000000A;
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-selected {
  background-color: #00000014;
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-2.Mui-selected:hover {
    background-color: #00000014;
  }
}

.emotion-2.Mui-selected.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-2.Mui-selected.Mui-disabled {
  opacity: 1;
  color: #00000042;
  background-color: #00000014;
}

.emotion-2.Mui-selected.Mui-disabled {
  color: #00000042;
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 1.25rem;
  margin: 0 -8px;
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-14 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  border-radius: 16px;
  text-align: center;
  box-sizing: border-box;
  min-width: 32px;
  padding: 0 6px;
  margin: 0 3px;
  color: #3B3B41DE;
  height: auto;
}

.emotion-14.Mui-disabled {
  opacity: 0.38;
}

<div>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-text emotion-0"
  >
    <ul
      class="MuiPagination-ul emotion-1"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext emotion-2"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon emotion-3"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          3
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          4
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          5
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <div
          class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis emotion-14"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 10"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page emotion-2"
          tabindex="0"
          type="button"
        >
          10
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext emotion-2"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon emotion-3"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
